pkg/graveler/ref/tag_iterator.go

```diff
@@ -8,26 +8,26 @@ import (
 )
 
 type KVTagIterator struct {
-	it     kv.MessageIterator
-	err    error
-	value  *graveler.TagRecord
-	repoID graveler.RepositoryID
-	store  kv.Store
-	ctx    context.Context
+	it    kv.MessageIterator
+	err   error
+	value *graveler.TagRecord
+	repo  *graveler.RepositoryRecord
```

Comment:
```suggestion
	partitionKey string
```
I don't see any use for the repository itself, other than getting the partition key

Reply:
👍🏽